src/core/math/arrayZipMap.ts

```diff
@@ -1,46 +1,67 @@
-import * as R from 'ramda';
-
 type Array<T> = T[];
 
 export function arrayMap<T1, TR>(
     a1: Array<T1>,
-    cb: (d1: T1, i: number) => TR
-) {
-    const mapArray = R.addIndex<T1, TR>(R.map);
+    fn: (d1: T1, i: number) => TR
+): Array<TR> {
+    const _a1_ = a1.length;
+
+    const res: Array<TR> = [];
 
-    return mapArray((iValue, i) => cb(iValue, i), a1);
+    for (let i = 0; i < _a1_; ++i) {
+        res.push(fn(a1[i], i));
```

Comment:
As long as we're hyperoptimizing, I think `const res = new Array(_a1_)` and `res[i] = fn(a1[i], i);` is a bit faster?

Reply:
Yes!